### PR TITLE
Remove deprecated get_course_catalog_api_client.

### PR DESCRIPTION
**Pull request by vkaracic**

### CODE REVIEW

**Comment by vkaracic** [comment]
@clintonb please review this when you get a chance.

File: ecommerce/coupons/tests/test_utils.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 
 @httpretty.activate
 @mock.patch(
-    'ecommerce.coupons.utils.get_course_catalog_api_client',
-    mock.Mock(return_value=EdxRestApiClient(
+    'ecommerce.core.models.SiteConfiguration.course_catalog_api_client',
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_utils.py]: This is copied seven times. Make a custom decorator for it, and put it in core/tests/decorators.py.

**Comment by clintonb** [comment]
:+1: once duplication resolved.

**Comment by vkaracic** [comment]
@clintonb decorator added!

**Comment by clintonb** [comment]
👍

**Pull request merged**